File: sample-dagger2-rx/src/main/java/com/hannesdorfmann/mosby/sample/dagger2/MainActivity.java
package com.hannesdorfmann.mosby.sample.dagger2;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.hannesdorfmann.mosby.sample.dagger2.repos.ReposFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.fragmentContainer, new ReposFragment())
                    .commit();
        }
    }
}

File: sample-flow/src/main/java/com/hannesdorfmann/mosby/sample/flow/MainActivity.kt
package com.hannesdorfmann.mosby.sample.flow

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.hannesdorfmann.mosby.sample.flow.countries.CountriesScreen
import com.hannesdorfmann.mosby.sample.flow.flow.AppDispatcher
import com.hannesdorfmann.mosby.sample.flow.flow.AppKeyParceler
import flow.Flow

class MainActivity : AppCompatActivity() {

  override fun attachBaseContext(baseContext: Context) {
    val newBase = Flow.configure(baseContext, this)
        .dispatcher(AppDispatcher(this))
        .defaultKey(CountriesScreen())
        .keyParceler(AppKeyParceler())
        .install()
    super.attachBaseContext(newBase)
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
  }

  override fun onBackPressed() {
    if (!Flow.get(this).goBack()) {
      super.onBackPressed();
    }
  }
}
